### PR TITLE
[Kernel][Expressions] Adds the NOT and IS_NOT_NULL expressions

### PR DESCRIPTION
**Pull request by allisonport-db**
#### Which Delta project/connector is this regarding?

- [ ] Spark
- [ ] Standalone
- [ ] Flink
- [X] Kernel
- [ ] Other (fill in here)

## Description

Adds the `NOT` and `IS NOT NULL` expressions.
- Adds them to the list of supported expressions in `Predicate`
- Adds support to the `DefaultExpressionEvaluator`

## How was this patch tested?

Adds unit tests.

### CODE REVIEW

File: kernel/kernel-api/src/main/java/io/delta/kernel/expressions/Predicate.java [hunk 1]
```diff
@@ -94,6 +107,17 @@ public Predicate(String name, List<Expression> children) {
         super(name, children);
     }
 
+    /** Constructor for a unary Predicate expression */
+    public Predicate(String name, Expression child) {
```
Comment (vkorukanti) [inline kernel/kernel-api/src/main/java/io/delta/kernel/expressions/Predicate.java:111]: Wondering if we need these two new constructors? It adds two more public APIs.

File: kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/DefaultExpressionEvaluator.java [hunk 2]
```diff
@@ -399,6 +415,29 @@ ColumnVector visitElementAt(ScalarExpression elementAt) {
...
+
+        @Override
+        ColumnVector visitIsNotNull(Predicate predicate) {
+            ColumnVector childResult = visit(childAt(predicate, 0));
+            int numRows = childResult.getSize();
+            boolean[] result = new boolean[numRows];
+            for (int rowId = 0; rowId < numRows; rowId++) {
+                result[rowId] = !childResult.isNullAt(rowId);
+            }
+            return new DefaultBooleanVector(numRows, Optional.empty(), result);
```
Comment (vkorukanti) [inline kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/DefaultExpressionEvaluator.java:438]: can we create a wrapper vector around the child vector?

```
return new ColumnVector() {
    getSize() { return childResult.getSize() }

    getBoolean() { return !childResult.getBoolean());

    ...
}
```

File: kernel/kernel-api/src/main/java/io/delta/kernel/expressions/Predicate.java (hunk 1 above)
Comment (vkorukanti) [reply kernel/kernel-api/src/main/java/io/delta/kernel/expressions/Predicate.java:111]: We could add utility methods to `internal.ExpressionUtils` to make the Predicate create simple.

**Review by vkorukanti — APPROVED** [review]
lgtm, couple of comments.

File: kernel/kernel-api/src/main/java/io/delta/kernel/expressions/Predicate.java (hunk 1 above)
Comment (allisonport-db) [reply kernel/kernel-api/src/main/java/io/delta/kernel/expressions/Predicate.java:111]: I can move it to the `ExpressionUtils` if you'd like (although I think we'd be needing these in `kernel-api` for the data skipping) but I also think these are okay public APIs to expose? They make sense and it's likely connectors would want to use these constructors anyways.

**Pull request merged**